components/automate-ui/src/app/pages/api-token/details/api-token-details.component.ts

```diff
@@ -98,4 +143,24 @@ export class ApiTokenDetailsComponent implements OnInit, OnDestroy {
   public get nameCtrl(): FormControl {
     return <FormControl>this.updateForm.controls.name;
   }
+
+  // updates whether the project was checked or unchecked
+  onProjectChecked(project: ProjectChecked): void {
+    this.projects[project.id].checked = project.checked;
+    const projectsSelected = Object.values(this.projects).filter(p => p.checked);
+
+    // since the app-projects-dropdown is not a true form input (select)
+    // we have to manage the form reactions
+    this.updateForm.controls.projects.setValue(projectsSelected.map(p => p.id));
+    this.updateForm.controls.projects.markAsDirty();
+  }
+
+  noProjectsUpdated(): boolean {
+    return xor(this.token.projects,
+      Object.values(this.projects).filter(p => p.checked).map(p => p.id)).length === 0;
```

Comment:
[nit] Same as above; I might be missing something 🤔 
```suggestion
      Object.keys(this.projects).filter(id => this.projects[id].checked).length === 0;
```